src/odemis/acq/align/fastem.py

```diff
@@ -18,22 +20,254 @@
 
+from fastem_calibrations.configure_hw import get_config_asm, configure_asm
 from odemis import model
-import time
 
+try:
+    from fastem_calibrations import (
+        autofocus_multiprobe,
+        scan_rotation_pre_align,
+        scan_amplitude_pre_align,
+        descan_gain,
+        image_translation_pre_align,
+        descan_gain,
+        image_rotation_pre_align,
+        image_rotation,
+        image_translation
+    )
+    fastem_calibrations = True
+except ImportError:
+    logging.info("fastem_calibrations package not found")
+    fastem_calibrations = False
+
+# TODO does it make sense to make this a list?
+OPTICAL_AUTOFOCUS = autofocus_multiprobe
+SCAN_ROTATION_PREALIGN = scan_rotation_pre_align
+SCAN_AMPLITUDE_PREALIGN = scan_amplitude_pre_align
+DESCAN_GAIN_STATIC = descan_gain
+IMAGE_ROTATION_PREALIGN = image_rotation_pre_align
+IMAGE_TRANSLATION_PREALIGN = image_translation_pre_align
+IMAGE_ROTATION_FINAL = image_rotation
+IMAGE_TRANSLATION_FINAL = image_translation
+
+# The executor is a single object, independent of how many times the module (fastem.py) is loaded.
 _executor = model.CancellableThreadPoolExecutor(max_workers=1)
 
 
-def align(main_data):
+def align(scanner, multibeam, descanner, detector, stage, ccd, beamshift, det_rotator, calibrations):
     """
-    :param main_data (odemis.gui.model.FastEMMainGUIData):
-    :returns: (ProgressiveFuture): acquisition future
+    Start a calibration task for a given list of calibrations.
+
+    :param scanner: (xt_client.Scanner) Scanner component connecting to the XT adapter.
+    :param multibeam: (technolution.EBeamScanner) The multibeam scanner component of the acquisition server module.
+    :param descanner: (technolution.MirrorDescanner) The mirror descanner component of the acquisition server module.
+    :param detector: (technolution.MPPC) The detector object to be used for collecting the image data.
+    :param stage: (actuator.ConvertStage) The stage in the corrected scan coordinate system, the x and y axes are
+        aligned with the x and y axes of the multiprobe and the multibeam scanner.
+    :param ccd: (model.DigitalCamera) A camera object of the diagnostic camera.
+    :param beamshift: (tfsbc.BeamShiftController) Component that controls the beamshift deflection.
+    :param det_rotator: (tmcm.CANController) K-mirror controller.
+    :param calibrations: (list of str) List of calibrations that should be run.
+
+    :returns: (ProgressiveFuture) Alignment future object, which can be cancelled.
+            The result of the future is: (None)
     """
+
+    if not fastem_calibrations:
+        raise ModuleNotFoundError("fastem_calibration module missing. Cannot run calibrations.")
+
     f = model.ProgressiveFuture()
-    _executor.submitf(f, _run_fake_alignment)
+
+    # Create a task that runs the calibration and alignments.
+    task = CalibrationTask(scanner, multibeam, descanner, detector, stage, ccd, beamshift, det_rotator, f, calibrations)
```

Comment:
You should estimate the duration and store it in the future either here, or in the init of the CalibrationTask. This way, even if the task is not yet executed (eg, executor is a little slow, or doing something else) it's possible for the GUI to know how long this task will take.

Reply:
Then we should also change this in the acquisition manager right?